ducktape/cluster/vagrant.py

```diff
@@ -24,10 +28,53 @@ class VagrantCluster(JsonCluster):
     advertised, this assumes that the Vagrant VM's name is a routeable hostname on all the hosts.
     """
 
-    def __init__(self):
-        hostname, ssh_hostname, username, flags = None, None, None, ""
-        nodes = []
+    def __init__(self, *args, **kwargs):
+        is_read_from_file = False
 
+        cluster_file = kwargs.get("cluster_file")
+        if cluster_file is not None:
+            try:
+                cluster_json = json.load(open(os.path.abspath(cluster_file)))
+                is_read_from_file = True
+            except IOError:
+                # It is OK if file is not found. Call vagrant ssh-info to read the cluster info.
+                pass
+
+        if not is_read_from_file:
+            cluster_json = {
+                "nodes": self._get_nodes_from_vagrant()
+            }
+
+        super(VagrantCluster, self).__init__(cluster_json)
+
+        # Load externally_routable_ip for each account
+        if is_read_from_file:
+            externally_routable_ip = cluster_json["externally_routable_ip"]
```

Comment:
@granders Good point. This is very reasonable. 

However, since you mentioned this, I find it a bit weird that `externally_routable_ip` is assigned in VagrantCluster but not in JsonCluster. Does this suggest that the existing implementation of JsonCluster doesn't actually work because it doesn't assign value to `RemoteAccount.externally_routable_ip`?